runtime/src/installed_scheduler_pool.rs

```diff
@@ -23,14 +27,40 @@ use {mockall::automock, qualifier_attr::qualifiers};
     allow(unused_attributes, clippy::needless_lifetimes)
 )]
 pub trait InstalledScheduler: Send + Sync + Debug + 'static {
+    // Calling this is illegal as soon as wait_for_termination is called.
     fn schedule_execution<'a>(
         &'a self,
         transaction_with_index: &'a (&'a SanitizedTransaction, usize),
     );
+
+    #[must_use]
+    fn wait_for_termination(&mut self, reason: &WaitReason) -> Option<ResultWithTimings>;
```

Comment:
Could you add some comment here about the intended benhavior?

Obviously its' impl dependent, but when should this return `None` vs `Some`? What if the bank terminates/pauses for a different reason?

EDIT:
I think adding a comment here will also help clear up some of the other questions I posted.

Reply:
did my best: https://github.com/solana-labs/solana/commit/507c5023f35b4b435c5fce9493c345ed924f9c60